### PR TITLE
Mait

### PR DESCRIPTION
**Pull request by jaanonim**
-endpoints
-validation
-jose token

Required: sceret_key_token.txt - it doesn't auto-generate

### CODE REVIEW

**Comment by Vakme** [comment]
Testy failują, przed merge trzeba będzie je poprawić

File: api_server/views.py [hunk 1]
```diff
@@ -1,10 +1,78 @@
...
+        # -------------if is no parameters trow all models------------------
+        if x==None and y==None:
+            try:
+                points = Point.objects.all()
+            except Point.DoesNotExist:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            serializer = PointSerializer(points, many=True)
+                
+            return Response(serializer.data)
```
Comment (Vakme) [inline api_server/views.py]: This part could be moved to separate function

File: api_server/urls.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from django.urls import include, path
+from api_server.views import getByID,getByPos,getTypeByID,postPoint,postType
+
+urlpatterns = [
+    path('points/', getByPos),
+    path('points/add/', postPoint),
+    path('points/<int:pk>/', getByID),
+    path('type-of-point/<int:pk>/', getTypeByID),
```
Comment (Vakme) [inline api_server/urls.py]: this path should be: `points/types/` or `types/` instead of `type-of-point/`

File: api_project/settings.py [hunk 3]
```diff
@@ -35,12 +37,25 @@
...
     SECRET_KEY_TOKEN = SECRET_KEY_TOKEN_FILE.read()
     SECRET_KEY_TOKEN_FILE.close()
 else:
-    print('ERROR: no secret_key_token.txt')
+    SECRET_KEY_FILE = open("./api_project/secret_key_token.txt", "w+")
+    SECRET_KEY_FILE.write(generateTokenKey())
+    SECRET_KEY_FILE.close()
+    SECRET_KEY_FILE = open("./api_project/secret_key_token.txt", "r")
+    SECRET_KEY = SECRET_KEY_FILE.read()
+    SECRET_KEY_FILE.close()
```
Comment (Wuzado) [inline api_project/settings.py]: You're overwriting the SECRET_KEY value!

File: api_project/settings.py [hunk 4]
```diff
@@ -35,12 +37,25 @@
...
+
+
```
Comment (Wuzado) [inline api_project/settings.py]: Unnecessary line breaks

File: api_server/views.py [hunk 5]
```diff
@@ -11,17 +11,10 @@ def getByPos(request, format=None):
         x=request.GET.get('x')
         y=request.GET.get('y')
 
-        # -------------if is no parameters trow all models------------------
+        # if is no parameters trow all models
```
Comment (Wuzado) [inline api_server/views.py]: Typo, should be: If no parameters are given, throw all models

File: api_project/settings.py [hunk 6]
```diff
@@ -12,6 +12,8 @@
 
 import os
 from os import path
+import random
+import string
```
Comment (Wuzado) [inline api_project/settings.py]: If you're switching to get_random_secret_key(), don't forget to remove these imports!

File: api_server/models.py [hunk 7]
```diff
@@ -21,25 +23,39 @@ class Point(models.Model):
...
-    vision_imp = models.IntegerField(blank=True, default='0')
-    vision_imp_desc = models.TextField(blank=True, default='')
-    mental_disorder = models.IntegerField(blank=True, default='0')
-    mental_disorder_desc = models.TextField(blank=True, default='')
-    staff = models.IntegerField(blank=True, default='0')
-    staff_desc = models.TextField(blank=True, default='')
 
     class Meta:
         ordering = ['ID']
+
```
Comment (Wuzado) [inline api_server/models.py]: Formatting (double line break)

File: api_server/views.py [hunk 8]
```diff
@@ -1,10 +1,81 @@
...
-    """
-    API endpoint that allows groups to be viewed or edited.
-    """
-    queryset = Point.objects.all().order_by('-ID')
-    serializer_class = PointSerializer
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.decorators import api_view, permission_classes
+from api_server.models import Point , TypeOfPoint
+from api_server.serializers import PointSerializerDetail , TypeOfPointSerializer, PointSerializer
```
Comment (Wuzado) [inline api_server/views.py]: Formatting (spaces before commas)

File: api_server/views.py [hunk 9]
```diff
@@ -1,10 +1,81 @@
...
+       
+
+        points = Point.objects.get(latitude=x,longitude=y)
+
+    except Point.DoesNotExist:
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+    serializer = PointSerializer(points)
+
+    return Response(serializer.data)
```
Comment (Wuzado) [inline api_server/views.py]: Such an odd way to line break here makes the file a bit unreadable. The code does not really appear as a visual block now.

Comment (Wuzado) [reply api_server/views.py]: This kind of formatting appears in the whole file. Could you reformat the whole file?

File: api_server/urls.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
+from django.urls import include, path
+from api_server.views import getByID,getByPos,getTypeByID,postPoint,postType
```
Comment (Wuzado) [inline api_server/urls.py]: Seems cluttery, could we separate them by spaces?

File: api_server/views.py [hunk 11]
```diff
@@ -1,10 +1,81 @@
...
+    if serializer.is_valid():
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+@api_view(['POST'])
+@permission_classes([canCreatePoint])
+def postType(request, format=None):
+
+    serializer =  TypeOfPointSerializer(data=request.data, context={'request': request})
```
Comment (Wuzado) [inline api_server/views.py]: Formatting (double space)

File: generateToken.py [hunk 12]
```diff
@@ -0,0 +1,12 @@
+from jose import jwt
+import os
+
+SECRET_KEY_TOKEN_FILE = open("./api_project/secret_key_token.txt", "r")
+SECRET_KEY_TOKEN = SECRET_KEY_TOKEN_FILE.read()
+SECRET_KEY_TOKEN_FILE.close()
+
```
Comment (Wuzado) [inline generateToken.py]: Unnecessary line break

File: api_server/tests.py [hunk 13]
```diff
@@ -1,7 +1,3 @@
 from django.test import TestCase
 
-from .models import Point
-
-class QuestionModelTests(TestCase):
-    def test(self):
-
```
Comment (Wuzado) [inline api_server/tests.py]: Well, I don't think deleting a whole test suite counts as fixing tests 😄

**Comment by Wuzado** [comment]
If you need to change the workflow of CircleCI (token generation or whatever you need) for tests, the config file is in .circleci directory.

**Comment by Wuzado** [comment]
Also, could you briefly describe backward incompatible changes, if any?

File: api_project/settings.py [hunk 14]
```diff
@@ -35,12 +37,25 @@
     SECRET_KEY_FILE.close()
 
 # secret key for token decode
+def generateTokenKey():
+    s=''
+    for x in range(0, 50):
+        s+=random.choice(string.ascii_lowercase)
+    return s
+
```
[the remark this answers is not in the document]
Comment (Wuzado) [reply api_project/settings.py]: Great idea with separate getKey function! KUDOS to you.

File: api_project/settings.py (hunk 3 above)
Comment (Wuzado) [reply api_project/settings.py]: Solved with separate function.

**Pull request merged**